src/control/createIndex.ts

```diff
@@ -98,52 +85,134 @@ export interface CreateIndexPodSpec {
+    if (options) {
+      ValidateProperties(options, [
+        'spec',
+        'name',
+        'dimension',
+        'metric',
+        'deletionProtection',
+        'waitUntilReady',
+        'suppressConflicts',
+      ]);
+    }
+    if (!options) {
+      throw new PineconeArgumentError(
+        'You must pass an object with required properties (`name`, `dimension`, `spec`) to create an index.'
+      );
+    }
+    if (!options.name || options.name.length === 0) {
+      throw new PineconeArgumentError(
+        'You must pass a non-empty string for `name` in order to create an index.'
+      );
+    }
+    if (!options.dimension || options.dimension <= 0) {
+      throw new PineconeArgumentError(
+        'You must pass a positive integer for `dimension` in order to create an index.'
+      );
+    }
+    if (!options.spec) {
+      throw new PineconeArgumentError(
+        'You must pass a pods or serverless `spec` object in order to create an index.'
+      );
+    }
+    if (options.spec) {
+      ValidateProperties(options.spec, ['serverless', 'pod']);
+    }
+    if (options.spec.serverless) {
+      ValidateProperties(options.spec.serverless, ['cloud', 'region']);
+      if (!options.spec.serverless.cloud) {
+        throw new PineconeArgumentError(
+          'You must pass a `cloud` for the serverless `spec` object in order to create an index.'
+        );
+      }
+      if (!options.spec.serverless.region) {
+        throw new PineconeArgumentError(
+          'You must pass a `region` for the serverless `spec` object in order to create an index.'
+        );
+      }
+    }
+    if (options.spec.pod) {
+      ValidateProperties(options.spec.pod, [
+        'environment',
+        'podType',
+        'replicas',
+        'shards',
+        'pods',
+        'metadataConfig',
+        'sourceCollection',
+      ]);
+      if (!options.spec.pod.environment) {
+        throw new PineconeArgumentError(
+          'You must pass an `environment` for the pod `spec` object in order to create an index.'
+        );
+      }
+      if (!options.spec.pod.podType) {
+        throw new PineconeArgumentError(
+          'You must pass a `podType` for the pod `spec` object in order to create an index.'
+        );
+      }
+    }
+    if (
+      options.spec.serverless &&
+      options.spec.serverless.cloud &&
+      !Object.values(ServerlessSpecCloudEnum).includes(
+        options.spec.serverless.cloud
+      )
+    ) {
+      throw new PineconeArgumentError(
+        `Invalid cloud value: ${
+          options.spec.serverless.cloud
+        }. Valid values are: ${Object.values(ServerlessSpecCloudEnum).join(
+          ', '
+        )}.`
+      );
+    }
+    if (
+      options.metric &&
+      !Object.values(IndexModelMetricEnum).includes(options.metric)
+    ) {
+      {
+        throw new PineconeArgumentError(
+          `Invalid metric value: ${options.metric}. Valid values are: 'cosine', 'euclidean', or 'dotproduct.'`
```

Comment:
We may have some inconsistencies in string formatting in some of these errors. Here we're denoting fields with `''` but we've used backticks elsewhere. Do we want to standardize that?

Reply:
Yeah, what I want to happen is: 
- Field names have backticks
- String values have single quotation marks

So in this instance, metric would look like ``metric``, and cosine would look like `'cosine'`